tests/unit/test_multipan_rcp.cpp

```diff
@@ -0,0 +1,795 @@
+        endFrame("Read Status Frame");
+    }
+
+    void enableRawLink()
+    {
+        static const bool isLinkEnabled = true;
+        createLinkEnableFrame(isLinkEnabled);
+        sendToRcp();
+    }
+
+    void disableRawLink()
+    {
+        static const bool isLinkEnabled = false;
+        createLinkEnableFrame(isLinkEnabled);
+        sendToRcp();
+    }
+
+    spinel_status_t startTransmit()
+    {
+        static const bool getResponse = true;
+        createTransmitFrame();
+        sendToRcp(getResponse);
+        return static_cast<spinel_status_t>(mNcp->getSpinelStatus());
+    };
+
+    void setScanChannelMask(uint32_t aMask)
+    {
+        createScanChannelMaskFrame(aMask);
+        sendToRcp();
+    }
+
+    uint32_t startEnergyScan()
+    {
+        static const bool getResponse = true;
+        createMacScanFrame();
+        sendToRcp(getResponse);
+        return mNcp->getSpinelStatus();
+    }
+
+    void getCommandStatus()
+    {
+        createReadStatusFrame();
+        sendToRcp();
+    }
+
+    void finishTransmit()
+    {
+        /* Reset instance submac state to sleep by resetting link
+           This is needed for a second transmit command to succeed
+           as the HandleTimer method will not be called to reset the submac */
+        disableRawLink();
+        enableRawLink();
+
+        /* Proceed with transmit done callback from ncp */
+        mNcp->processTransmit();
+    };
+
+private:
+    void startFrame(unsigned int aCommand, spinel_prop_key_t aKey)
+    {
+        uint8_t spinelHeader = SPINEL_HEADER_FLAG | mIid | mTid;
+
+        SuccessOrQuit(mEncoder.BeginFrame(Spinel::Buffer::kPriorityLow));
+        SuccessOrQuit(mEncoder.WriteUint8(spinelHeader));
+        SuccessOrQuit(mEncoder.WriteUintPacked(aCommand));
+        SuccessOrQuit(mEncoder.WriteUintPacked(aKey));
+    }
+
+    void endFrame(const char *aTextMessage)
+    {
+        static const uint16_t display_length = 64;
+        SuccessOrQuit(mEncoder.EndFrame());
+        // DumpBuffer(aTextMessage, mBuf, display_length);
+    }
+
+    void sendToRcp(bool needsResponse = false)
+    {
+        static const uint8_t data_offset     = 2;
+        bool                 requestResponse = false;
+
+        size_t frame_len = mBuffer.OutFrameGetLength();
+
+        mOffset += data_offset;
+
+        mNcp->Receive(mBuf + mOffset, frame_len);
+
+        /* Some spinel commands immediately send queued responses when command is complete
+           while others require a separate command to the ncp in order to receive the response.
+           If a response is needed and not immediately received. Issue a command to update the status. */
+        requestResponse = needsResponse && !mNcp->gotResponse(mIid, mTid);
+
+        mTid = SPINEL_GET_NEXT_TID(mTid);
+        SuccessOrQuit(mBuffer.OutFrameRemove());
+
+        mOffset += frame_len;
+        mOffset %= kTestBufferSize;
+
+        if (requestResponse)
+        {
+            getCommandStatus();
```

Comment:
Is this going to be a recursive call?

Reply:
It would not be recursive. This will send off at most one more frame through sendToRcp which will not request a response.